src/hooks/sync-sites/use-sync-push.ts

```diff
@@ -0,0 +1,169 @@
+import * as Sentry from '@sentry/electron/renderer';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import { useCallback, useEffect, useMemo } from 'react';
+import { getIpcApi } from '../../lib/get-ipc-api';
+import { useAuth } from '../use-auth';
+import { SyncSite } from '../use-fetch-wpcom-sites';
+import { useSyncStatesProgressInfo, PushStateProgressInfo } from '../use-sync-states-progress-info';
+import { usePullPushStates } from './use-pull-push-states';
+
+export type SyncPushState = {
+	remoteSiteId: number;
+	status: PushStateProgressInfo;
+	selectedSite: SiteDetails;
+	isStaging: boolean;
+};
+
+export function useSyncPush( {
+	pushStates,
+	setPushStates,
+}: {
+	pushStates: Record< string, SyncPushState >;
+	setPushStates: React.Dispatch< React.SetStateAction< Record< string, SyncPushState > > >;
+} ) {
+	const { __ } = useI18n();
+	const { client } = useAuth();
+	const {
+		updateState: updatePushState,
+		getState,
+		clearState: clearPushState,
+	} = usePullPushStates< SyncPushState >( pushStates, setPushStates );
+	const { pushStatesProgressInfo, isKeyPushing, isKeyFinished, isKeyFailed } =
+		useSyncStatesProgressInfo();
+
+	const getPushProgressInfo = useCallback(
+		async ( selectedSiteId: string, remoteSiteId: number ) => {
+			if ( ! client ) {
+				return;
+			}
+
+			const response = await client.req.get< {
+				status: 'finished' | 'failed' | 'initial_backup_started' | 'archive_import_started';
+				success: boolean;
+			} >( {
+				path: `/sites/${ remoteSiteId }/studio-app/sync/import`,
+				apiNamespace: 'wpcom/v2',
+			} );
+
+			let status: PushStateProgressInfo = pushStatesProgressInfo.importing;
+			if ( response.success && response.status === 'finished' ) {
+				status = pushStatesProgressInfo.finished;
+			} else if ( response.success && response.status === 'failed' ) {
+				status = pushStatesProgressInfo.failed;
+			}
+			// Update state in any case to keep polling push state
+			updatePushState( selectedSiteId, remoteSiteId, {
+				status,
+			} );
+		},
+		[ client, pushStatesProgressInfo, updatePushState ]
+	);
+
+	const pushSite = useCallback(
+		async ( connectedSite: SyncSite, selectedSite: SiteDetails ) => {
+			if ( ! client ) {
+				return;
+			}
+			const remoteSiteId = connectedSite.id;
+			updatePushState( selectedSite.id, remoteSiteId, {
+				remoteSiteId,
+				status: pushStatesProgressInfo.creatingBackup,
+				selectedSite,
+				isStaging: connectedSite.isStaging,
+			} );
+
+			const { archiveContent, exceedsSizeLimit } = await getIpcApi().archiveSite(
+				selectedSite.id,
+				'tar'
+			);
+			if ( exceedsSizeLimit ) {
+				getIpcApi().showErrorMessageBox( {
+					title: sprintf( __( 'Error pushing to %s' ), connectedSite.name ),
+					message: __(
+						'The site is too large to push. Please reduce the size of the site and try again.'
+					),
+				} );
```

Comment:
I'm displaying this generic error. I could also display the error that comes from the backend.

When trying to upload a bakcup bigger than the limit, we cannot grab the actual error message, because we receive a CORS error.